### DISCUSSION TITLE
Képnormalizáció

### DISCUSSION THREAD

**Discussion by qbeer** [csabaiBio/ai4covid#9] [Ideas]
https://stackoverflow.com/questions/70993502/detect-if-an-image-is-negative

**Comment by qbeer** [comment]
@bsulyok nekem eddig ez működik a legjobban:

```python
def normalize_image(self, image):
        img = np.clip(np.float32(image), np.quantile(image, 0.02), np.quantile(image, 0.98))
        print(np.min(img), np.max(img), np.mean(img), np.std(img))
        img = (img - np.min(img))/(np.max(img) - np.min(img)) # min-max scaling
        img = (img * 255).astype('uint8')
        print(np.min(img), np.max(img), np.mean(img), np.std(img))
        clahe = cv2.createCLAHE(clipLimit=0.01, tileGridSize=(8, 8))
        img = clahe.apply(img)
        print(np.min(img), np.max(img), np.mean(img), np.std(img))
        img = cv2.medianBlur(img, 3)
        print(np.min(img), np.max(img), np.mean(img), np.std(img))
        print('*' * 100)
        img = np.array(img) / 255.
        return img
```

**Reply by bsulyok** [reply]
a quantile szerinti clip-elés jó lesz, lehet, ez a hiányzó elem a preprocessing-hez...

**Reply by bsulyok** [reply]
Megtaláltam az "is_inverted" függvény ideális paramétereit, ezzel az 1103 képből csak egynél nem tudja helyesen megítélni, hogy invertálva van-e a kép, a paraméterek és ez a nehéz képet a notebook-ban látható.

**Comment by bsulyok** [comment]
Az invertáltság detektálása 100%-on működik a teszt képeken, a 'P_3_335.png' kép el van forgatva.

**Reply by bsulyok** [reply]
A train képek között pedig ezek vannak elforgatva: 'P_1_60.png', 'P_1_163.png', 'P_694.png', 'P_829.png'.
